frontend/app/components/Dashboard/Widgets/CustomMetricsWidgets/CustomMetricPieChart/CustomMetricPieChart.tsx

```
@@ -1,126 +1,156 @@
-            delete filter.operatorOptions
-            delete filter.category
-            delete filter.icon
-            delete filter.label
-            delete filter.options
+  const onClickHandler = (event) => {
+    if (event && !event.payload.group) {
+      const filters = Array<any>();
+      let filter = { ...filtersMap[metric.metricOf] };
+      filter.value = [event.payload.name];
+      filter.type = filter.key;
+      delete filter.key;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)